metrics/web/docs/features/navigation/01_navigation_design.md

```diff
@@ -126,15 +126,24 @@ class MetricsRouterDelegate extends RouterDelegate<RouteConfiguration>
 
-The `NavigationNotifier` provides several navigation methods that are similar to methods the `Navigator` widget provides. But unlike the `Navigator` methods, the ones of `NavigationNotifier` do not return a `Future`. The main reason is that the [Metrics Web presentation layer architecture](#https://github.com/platform-platform/monorepo/blob/master/metrics/web/docs/02_presentation_layer_architecture.md) declares the interactions between pages and the UI logic to be in a presenter (`ChangeNotifier`, in our case) and performed using `View Models`. Thus, the following statements hold:
+The `NavigationNotifier` provides several navigation methods that are similar to methods the `Navigator` widget provides. But unlike the `Navigator` methods, the ones of `NavigationNotifier` do not return a `Future`. The main reason is that the [Metrics Web presentation layer architecture](https://github.com/platform-platform/monorepo/blob/master/metrics/web/docs/02_presentation_layer_architecture.md) declares the interactions between pages and the UI logic to be in a presenter (`ChangeNotifier`, in our case) and performed using `View Models`. Thus, the following statements hold:
 - using the `MetricsPage` data directly in the UI (for example, in a page widget) or adding such logic to the widget itself - breaks the principle of separation between the presentation layer components and complexifies testing;
 - implementing the data flow (using `View Model`s) between pages/widgets using the result param of the `.pop()` method - breaks the principle of separation between the presentation layer components and complexifies testing.
 
 According to the above, we **do not** pass parameters using internal navigation and **do not** implement similar logic for the new navigation system within the Metrics Web Application.
 
+To use the application correctly, all of its components, such as `Feature Config` or `Local Config`, must be initialized. The application must show the `Loading Page` when the initialization is in progress. Consider the following sequence diagram that describes the strategy to use in navigation when the application is not initialized.
+
+![Not Initialized App Navigation Sequence Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://github.com/platform-platform/monorepo/raw/update_navigator_designs/metrics/web/docs/features/navigation/diagrams/not_initialized_app_navigation_sequence_diagram.puml)
+
+1. When the user enters the application, the `_isAppInitialized` variable of the `NavigationNotifier` is `false`.
+2. The `handleInitialRoutePath` saves the given initial route configuration in the `_redirectRoute` variable and pushes the `LoadingPage`.
+3. When the initialization finishes, the application calls `handleAppInitializationUpdates(isAppInitialized: true)`.
+4. `NavigationNotifier` sets the `_isAppInitialized` value to `true` and redirects user to the `_redirectRoute`.
```

Comment:
I think this part becomes redundant if we'll improve the sequence diagram as we discussed in slack.